pkg/kube/kclient/client.go

```diff
@@ -310,17 +310,19 @@ func newInformerClient[T controllers.ComparableObject](inf informerfactory.Start
 		informer:      inf.Informer,
 		startInformer: inf.Start,
 	}
-	applyDynamicFilter(filter, ic)
+	if filter.ObjectFilter != nil {
+		applyDynamicFilter(filter, types.GetGVR[T](), ic)
```

Comment:
types.GetGVR[T] panic, here the T is a interface. 

We are using controllers.Object around, ptr.Empty of a interface returns nil.

Reply:
the case where it can panic is only for the delayedInformer code path; this code path is fine. it always uses concrete type. the delayedInformer has user explicitly pass gvr

Reply:
Yes, but the delayed informer calls newInformerClient innnerly

Reply:
Good call, my test was not robust enough. Added a test that fails now, and fixed the failure